libs/ml/examples/word2vec_distributed_learning/main.cpp

```diff
@@ -0,0 +1,173 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/random.hpp"
+#include "file_loader.hpp"
+#include "math/clustering/knn.hpp"
+#include "math/matrix_operations.hpp"
+#include "math/statistics/mean.hpp"
+#include "math/tensor.hpp"
+#include "ml/core/graph.hpp"
+#include "ml/dataloaders/word2vec_loaders/sgns_w2v_dataloader.hpp"
+#include "ml/distributed_learning/coordinator.hpp"
+#include "ml/layers/fully_connected.hpp"
+#include "ml/layers/skip_gram.hpp"
+#include "ml/ops/activation.hpp"
+#include "ml/ops/loss_functions.hpp"
+#include "ml/optimisation/sgd_optimiser.hpp"
+#include "model_saver.hpp"
+#include "word2vec_client.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstddef>
```

Comment:
i suspect some of these includes are surplus to requirements.
please find out which and remove them